specification/logic/resource-manager/Microsoft.Logic/stable/2016-06-01/logic.json

```diff
@@ -9030,6 +9398,320 @@
         "modelAsString": false
       }
     },
+    "IntegrationAccountRosettaNetProcessConfigurationListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IntegrationAccountRosettaNetProcessConfiguration"
+          },
+          "description": "The list of integration account RosettaNet PIP configuration."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results."
+        }
+      },
+      "description": "The list of integration account RosettaNet process configuration."
+    },
+    "IntegrationAccountRosettaNetProcessConfiguration": {
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/IntegrationAccountRosettaNetProcessConfigurationProperties",
+          "description": "The integration account RosettaNet process configuration properties."
+        }
+      },
+      "description": "The integration account RosettaNet process configuration.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ]
+    },
+     "IntegrationAccountRosettaNetProcessConfigurationProperties": {
+      "type": "object",
+      "required": [
+        "processCode",
+        "processVersion",
+        "processName",
+        "activitySettings",
+        "initiatorRoleSettings",
+        "responderRoleSettings"
+      ],
+      "properties": {
+        "createdTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "The created time."
+        },
+        "changedTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "The changed time."
+        },
+        "metadata": {
+          "type": "object",
+          "description": "The metadata.",
+          "properties": {}
+        },
```

Comment:
This is pretty much unusable in an SDK because it will generate as an object in c# for example. Is this arbitrary data or can there be a schema for this? If this supports key value pairs where value is of type string, you could use something like this:
```
"additionalProperties": {
  "type": "string"
}
```

Reply:
updated in new iteration

Reply:
Sorry. That's not entirely correct. I didn't mean change type to string. You should keep type `object` and add the extra declaration like this: [https://github.com/Azure/azure-rest-api-specs/blob/8b5cb0ff10bca2b179cdc9c7fd8c13336494a19e/specification/eventhub/resource-manager/Microsoft.EventHub/stable/2017-04-01/EventHub.json#L2047](https://github.com/Azure/azure-rest-api-specs/blob/8b5cb0ff10bca2b179cdc9c7fd8c13336494a19e/specification/eventhub/resource-manager/Microsoft.EventHub/stable/2017-04-01/EventHub.json#L2047)

---
In reply to: [309005645](https://github.com/Azure/azure-rest-api-specs/pull/6702#discussion_r309005645) [](ancestors = 309005645)